modules/xfem/src/base/XFEM.C

```diff
@@ -967,6 +989,31 @@ XFEM::cutMeshWithEFA()
 
       libmesh_elem->set_node(j) = libmesh_node;
 
+      // Store solution for all nodes affected by XFEM (even existing nodes)
+      if (parent_elem->is_semilocal(_mesh->processor_id()))
+      {
+        Node * solution_node = libmesh_node; //Node from which to store solution
+
+        for (auto nn : NewNodes)
+        {
+          if (nn->id() == libmesh_node->id())
+          {
+            solution_node = _mesh->node_ptr(nn->parent()->id());
+            break;
+          }
+        }
+
+        if ((_moose_mesh->isSemiLocal(solution_node)) ||
```

Comment:
Hmm... This may not be correct. isSemiLocal just means that the element is attached to a local element. libMesh has better methods for testing for ownership, ghosting, evaluable, these days.